### PR TITLE
Driver for altimeter functionality of Air Control Display

### PR DESCRIPTION
**Pull request by Scumi**
Driver for the altimeter functionality of the Air Avionics - Air Control Display in accordance with the Installation Manual 2.0. Driver has been tested on my Openvario.

Ref.: #161

### CODE REVIEW

File: src/Device/Driver/AirControlDisplay.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  /* virtual methods from class Device */
+  bool ParseNMEA(const char *line, struct NMEAInfo &info) override;
+  bool PutQNH(const AtmosphericPressure &pres,
+              OperationEnvironment &env) override;
+};
+
+bool
+ACDDevice::PutQNH(const AtmosphericPressure &pres, OperationEnvironment &env)
+{
+  if (!EnableNMEA(env))
```
Comment (MaxKellermann) [inline src/Device/Driver/AirControlDisplay.cpp]: Why this call?

File: build/driver.mk [hunk 2]
```diff
@@ -124,6 +124,7 @@ DRIVER_SOURCES = \
 	$(DRIVER_SRC_DIR)/Zander.cpp \
 	$(DRIVER_SRC_DIR)/Vaulter.cpp \
 	$(DRIVER_SRC_DIR)/KRT2.cpp \
-	$(DRIVER_SRC_DIR)/ATR833.cpp
+	$(DRIVER_SRC_DIR)/ATR833.cpp \
+	$(DRIVER_SRC_DIR)/AirControlDisplay.cpp
```
Comment (MaxKellermann) [inline build/driver.mk]: Should be moved up to avoid modifying the ATR833 line in this patch

File: src/Device/Driver/AirControlDisplay.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+#include "NMEA/InputLine.hpp"
+#include "NMEA/Checksum.hpp"
+#include "Atmosphere/Pressure.hpp"
+#include "Units/System.hpp"
+#include "Math/Util.hpp"
+
+static bool
+ParsePAAVS(NMEAInputLine &line, NMEAInfo &info)
+{
+  double value;
```
Comment (MaxKellermann) [inline src/Device/Driver/AirControlDisplay.cpp]: According to your code comments, the values are integer. Why parse a `double` here?

File: src/Device/Driver/AirControlDisplay.cpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+              OperationEnvironment &env) override;
+};
+
+bool
+ACDDevice::PutQNH(const AtmosphericPressure &pres, OperationEnvironment &env)
+{
+  if (!EnableNMEA(env))
+    return false;
+  char buffer[100];
+  unsigned QNHinPascal = uround(pres.GetPascal());
```
Comment (MaxKellermann) [inline src/Device/Driver/AirControlDisplay.cpp]: Coding style: bad variable name

File: src/Device/Driver/AirControlDisplay.cpp [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+#include "NMEA/Checksum.hpp"
+#include "Atmosphere/Pressure.hpp"
+#include "Units/System.hpp"
+#include "Math/Util.hpp"
+
+static bool
+ParsePAAVS(NMEAInputLine &line, NMEAInfo &info)
+{
+  int value;
+
```
Comment (MaxKellermann) [inline src/Device/Driver/AirControlDisplay.cpp]: There are many lines like this with trailing whitespace.

File: build/driver.mk [hunk 6]
```diff
@@ -87,7 +87,7 @@ XCTRACER_SOURCES = \
 
 THERMALEXPRESS_SOURCES = \
 	$(DRIVER_SRC_DIR)/ThermalExpress/Driver.cpp
- 
+
```
Comment (MaxKellermann) [inline build/driver.mk]: This is an example of a line where we had a mistakenly added trailing space, and you removed it in this unrelated commit. This is confusing to read, and this is why there shouldn't be trailing whitespace. (See below for examples of trailing whitespace you added.)

**Comment by Scumi** [comment]
Your request for the unit test was a good idea. I recognized a mismatch between the documentation and the transferred NMEA sentences. The manufacturer confirmed the mismatch. I reverted your requested change of the variable declaration, but this time updated the helptext. The altimeter sends the altitude with two decimal places.
